impl/maven-xml/src/main/java/org/apache/maven/internal/xml/XmlNodeUtil.java

```diff
@@ -18,160 +18,28 @@
-        this.children = ImmutableCollections.copy(children);
-        this.location = location;
-    }
-
-    @Override
-    public XmlNode merge(XmlNode source, Boolean childMergeOverride) {
-        return merge(this, source, childMergeOverride);
-    }
-
-    // ----------------------------------------------------------------------
-    // Name handling
-    // ----------------------------------------------------------------------
-
-    @Override
-    public String getPrefix() {
-        return prefix;
-    }
-
-    @Override
-    public String getNamespaceUri() {
-        return namespaceUri;
-    }
-
-    @Override
-    public String getName() {
-        return name;
-    }
-
-    // ----------------------------------------------------------------------
-    // Value handling
-    // ----------------------------------------------------------------------
-
-    public String getValue() {
-        return value;
-    }
-
-    // ----------------------------------------------------------------------
-    // Attribute handling
-    // ----------------------------------------------------------------------
-
-    @Override
-    public Map<String, String> getAttributes() {
-        return attributes;
-    }
-
-    public String getAttribute(String name) {
-        return attributes.get(name);
-    }
-
-    // ----------------------------------------------------------------------
-    // Child handling
-    // ----------------------------------------------------------------------
-
-    public XmlNode getChild(String name) {
-        if (name != null) {
-            ListIterator<XmlNode> it = children.listIterator(children.size());
-            while (it.hasPrevious()) {
-                XmlNode child = it.previous();
-                if (name.equals(child.getName())) {
-                    return child;
-                }
-            }
-        }
-        return null;
-    }
-
-    public List<XmlNode> getChildren() {
-        return children;
-    }
-
-    public int getChildCount() {
-        return children.size();
-    }
-
-    // ----------------------------------------------------------------------
-    // Input location handling
-    // ----------------------------------------------------------------------
+import static org.apache.maven.api.xml.XmlNode.CHILDREN_COMBINATION_APPEND;
+import static org.apache.maven.api.xml.XmlNode.CHILDREN_COMBINATION_MODE_ATTRIBUTE;
+import static org.apache.maven.api.xml.XmlNode.ID_COMBINATION_MODE_ATTRIBUTE;
+import static org.apache.maven.api.xml.XmlNode.KEYS_COMBINATION_MODE_ATTRIBUTE;
+import static org.apache.maven.api.xml.XmlNode.SELF_COMBINATION_MODE_ATTRIBUTE;
+import static org.apache.maven.api.xml.XmlNode.SELF_COMBINATION_OVERRIDE;
+import static org.apache.maven.api.xml.XmlNode.SELF_COMBINATION_REMOVE;
 
-    /**
-     * @since 3.2.0
-     * @return input location
-     */
-    public Object getInputLocation() {
-        return location;
-    }
+public class XmlNodeUtil {
```

Comment:
I prefer the Guava convention of using plurals for utility classes; e.g. instead of XmlNodeUtil consider XmlNodes